python/utils.py

```diff
@@ -453,6 +453,15 @@ def _unloadPluginModules(packageName):
                 sys.modules[mod].qCleanupResources()
         except:
             pass
+
+        # try removing path
+        try:
+            if hasattr(sys.modules[mod], '__path__'):
+                for path in sys.modules[mod].__path__:
+                    sys.path.remove(path)
+        except:
```

Comment:
Can you explicitly list the exceptions which are caught in here?

Reply:
Eventually, if the path does not exists in `sys.path` it will raise a 'ValueError'.

Reply:
@m-kuhn do you want a explicit except clause ?

Reply:
In this case, would it make sense to move the try catch only around the remove to continue if it's only one of the paths that's missing?

And in general we should prefer explicit except clauses, I've seen errors masked by catch-all exceptions too many times :)

Reply:
Ok; in that line of thoughts, we can also add logging warning with stack trace to other catch-all exceptions around ?

Reply:
Yes. I like the idea.
I am a bit afraid of a bad signal to noise ratio because plenty of them probably were introduced to catch expected exceptions which will flood the log. So maybe opt-in (or debug build only) and we can incrementally get rid of them?

Reply:
Let's go with Qgis.Debug log level

Reply:
sounds good

Reply:
Changes done.

Reply:
mhhh there is no Qgis.Debug level :-/

Reply:
Maybe qDebug is more appropriate

Reply:
Anyone knows where qDebug output messages with Qgis desktop ?

Reply:
using qDebug instead QgsLogMessage